src/Catalog.API/Apis/CatalogApi.cs

```diff
@@ -52,21 +113,27 @@ public static async Task<Results<Ok<PaginatedItems<CatalogItem>>, BadRequest<str
         return TypedResults.Ok(new PaginatedItems<CatalogItem>(pageIndex, pageSize, totalItems, itemsOnPage));
     }
 
+    [ProducesResponseType<ProblemDetails>(StatusCodes.Status400BadRequest, "application/problem+json")]
     public static async Task<Ok<List<CatalogItem>>> GetItemsByIds(
         [AsParameters] CatalogServices services,
-        int[] ids)
+        [Description("List of ids for catalog items to return")] int[] ids)
     {
         var items = await services.Context.CatalogItems.Where(item => ids.Contains(item.Id)).ToListAsync();
         return TypedResults.Ok(items);
     }
 
-    public static async Task<Results<Ok<CatalogItem>, NotFound, BadRequest<string>>> GetItemById(
+    [ProducesResponseType<ProblemDetails>(StatusCodes.Status400BadRequest, "application/problem+json")]
+    public static async Task<Results<Ok<CatalogItem>, NotFound, BadRequest<ProblemDetails>>> GetItemById(
+        HttpContext httpContext,
         [AsParameters] CatalogServices services,
-        int id)
+        [Description("The catalog item id")] int id)
     {
         if (id <= 0)
         {
-            return TypedResults.BadRequest("Id is not valid.");
+            var problemDetails = problemDetailsFactory.CreateProblemDetails(httpContext,
+                detail: $"Id is not valid", statusCode: StatusCodes.Status400BadRequest);
+            httpContext.Response.ContentType = "application/problem+json";
```

Comment:
Sadly, the `CreateProblemDetails` method fails to set the content-type to "application/problem+json". Even worse is that _despite_ adding the explicit "ProducesResponseType" on this method with the correct content-type, this doesn't make it into the OpenAPI document -- probably because the BadRequest<ProblemDetails> overrides it (incorrectly IMHO).